client/client.go

```diff
@@ -123,6 +124,167 @@ func (c *Client) CheckAll(ctx context.Context, cs *consistency.Strategy, rs []re
+		if req.OptionalCursor != nil {
+			return fmt.Errorf("cursor already set in LookupResources request")
+		}
+
+		req.OptionalCursor = cursor
+		return nil
+	}
+}
+
+// WithLRCaveatContext configures the LookupResources request to include a caveat context.
+func WithLRCaveatContext(caveatCtx map[string]any) LookupResourcesOpts {
+	return func(req *v1.LookupResourcesRequest) error {
+		c, err := structpb.NewStruct(caveatCtx)
+		if err != nil {
+			return fmt.Errorf("failed to convert caveat context: %w", err)
+		}
+
+		req.Context = c
+		return nil
+	}
+}
+
+// FoundResource represents a resource found by the LookupResources method.
+//
+// It is up to the caller to determine if the resource was found absolutely,
+// or is a conditional result.
+type FoundResource struct {
+	resourceObjectID           string
+	nextResourceCursor         *v1.Cursor
+	missingCaveatContextFields []string
+}
+
+// ResourceObjectID returns the resource object ID of the found resource. The boolean
+// indicates whether the resource was found absolutely (i.e., no missing caveat context fields).
+// If the boolean is false, the resource was found conditionally and the caller
+// should not assume the resource exists without the caveat context being further specified.
+func (fr FoundResource) ResourceObjectID() (string, bool) {
+	return fr.resourceObjectID, len(fr.missingCaveatContextFields) == 0
+}
+
+// Cursor returns the cursor for the next resource in the LookupResources request,
+// to be passed via WithLRCursor for pagination.
+func (fr FoundResource) Cursor() *v1.Cursor {
+	return fr.nextResourceCursor
+}
+
+// MissingCaveatContextFields returns the list of caveat context fields that were
+// missing for the LookupResources request, for this resource.
+func (fr FoundResource) MissingCaveatContextFields() []string {
+	return fr.missingCaveatContextFields
+}
+
+// PaginatedLookupResources performs a lookup for resources that the subject has the
+// provided permission on, automatically handling pagination based on the given
+// page size.
+func (c *Client) PaginatedLookupResources(
+	ctx context.Context,
+	cs *consistency.Strategy,
+	resourceType string,
+	permission string,
+	subject rel.Subject,
+	pageSize uint32,
+	opts ...LookupResourcesOpts,
+) iter.Seq2[FoundResource, error] {
+	return func(yield func(FoundResource, error) bool) {
+		var currentCursor *v1.Cursor
+		for {
+			foundResourceCount := 0
+			updatedOpts := append([]LookupResourcesOpts{WithLRLimit(pageSize), WithLRCursor(currentCursor)}, opts...)
+			it := c.LookupResources(ctx, cs, resourceType, permission, subject, updatedOpts...)
+			for fr, err := range it {
+				if !yield(fr, err) {
+					return
+				}
+				foundResourceCount++
+				currentCursor = fr.Cursor()
+			}
+			if foundResourceCount < int(pageSize) {
+				break
+			}
+		}
+	}
+}
+
+// LookupResources performs a lookup for resources that the subject has the
+// provided permission on.
+func (c *Client) LookupResources(
```

Comment:
Is there ever a time you'd want to use this method if the paginated method exists?

Reply:
Yeah; if you want to control the pagination manually